src/components/Note/Note.jsx

```diff
@@ -2,31 +2,110 @@ import React from "react";
 import { useNotes } from "../../context/note-context";
 import "./Note.css";
 
-function Note({ _id, title, note, noteColor, pinned }) {
+function Note({ singleNote, trash, archive }) {
+  const { _id, title, note, noteColor, pinned } = singleNote;
   const { notesDispatch } = useNotes();
 
   const pinNoteHandler = () => {
     notesDispatch({ type: "TOGGLE_PIN_NOTE", payload: _id });
   };
 
+  const deleteNoteHandler = () => {
+    notesDispatch({ type: "DELETE_NOTE", payload: singleNote });
+  };
+
+  const archiveNoteHandler = (type) => {
+    if (type === "archive") {
+      notesDispatch({ type: "ARCHIVE_NOTE", payload: singleNote });
+    } else {
+      notesDispatch({ type: "UNARCHIVE_NOTE", payload: singleNote });
+    }
+  };
+
   return (
     <div className={`card m-1 p-2 note-color-${noteColor}`}>
       <div className="note-header">
-        <h3>{title}</h3>
-        {pinned ? (
-          <div className="note-icon center-div" title="Unpin note" onClick={pinNoteHandler}>
-            <i className="fas fa-thumbtack pin-note-icon"></i>
-          </div>
-        ) : (
-          <div className="note-icon center-div" title="Pin note" onClick={pinNoteHandler}>
-            <i className="fas fa-thumbtack unpin-note-icon"></i>
-          </div>
-        )}
+        <h3 className="note-title">{title}</h3>
+        {!trash &&
+          !archive &&
+          (pinned ? (
+            <div
+              className="note-icon center-div"
+              title="Unpin note"
+              onClick={pinNoteHandler}
+            >
+              <i className="fas fa-thumbtack pin-note-icon"></i>
+            </div>
+          ) : (
+            <div
+              className="note-icon center-div"
+              title="Pin note"
+              onClick={pinNoteHandler}
+            >
+              <i className="fas fa-thumbtack unpin-note-icon"></i>
+            </div>
+          ))}
```

Comment:
title of div and classname of icon is only getting changed right ?
Can you do ternary condition within title attribute of div and classname of icon ?
Why ternary condition for entire div element ?

Reply:
Make sense. thank you so much for suggestion